### PR TITLE
Switch to ESM

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

**Comment by flurdy** [comment]
Do we need the typescript config file?

File: package.json [hunk 1]
```diff
@@ -50,31 +51,36 @@
...
     "eslint-plugin-promise": "6.2.0",
     "husky": "9.0.11",
     "jest": "29.7.0",
-    "nodemon": "3.1.2",
+    "nodemon": "3.1.4",
     "npm-run-all": "4.1.5",
-    "pino-pretty": "11.1.0",
-    "prettier": "3.3.0",
-    "resolve-url-loader": "5.0.0"
+    "pino-pretty": "11.2.1",
```
Comment (flurdy) [inline package.json:83]: did we move this to non-dev dependency in other repos?

Comment (flurdy) [reply package.json:83]: (not specific to this change though)

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+module.exports = {
```
Comment (feedmypixel) [inline .eslintrc.cjs:1]: Looks like theres some bits missing from this config - can you update it to be the same as - https://github.com/DEFRA/cdp-node-backend-template/blob/main/.eslintrc.cjs

**Comment by feedmypixel** [comment]
Also can you update `jsconfig.json` like this - https://github.com/DEFRA/cdp-portal-frontend/pull/514/commits/6afd0d1af3fdc8eaa1140566368d1a9f7e673d2b

File: src/helpers/ms-graph.js [hunk 3]
```diff
@@ -49,6 +49,7 @@ const msGraphPlugin = {
         baseUrl: azureClientBaseUrl,
         ...(proxy && {
           fetchOptions: {
+            // @ts-expect-error This is not typed
```
Comment (feedmypixel) [inline src/helpers/ms-graph.js:52]: would be worth adding this link in here - https://github.com/microsoftgraph/msgraph-sdk-javascript/issues/1646#issuecomment-2200783014

File: src/index.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import process from 'node:process'
-import { createLogger } from '~/src/helpers/logging/logger'
-import { startServer } from '~/src/helpers/start-server'
+import { createLogger } from '~/src/helpers/logging/logger.js'
+import { startServer } from '~/src/helpers/start-server.js'
 
-startServer()
+await startServer()
```
Comment (feedmypixel) [inline src/index.js:5]: :100:

**Review by feedmypixel — APPROVED** [review]
LGTM - 1 comment about adding a link to the comment I made about the proxy dispatcher use.

Also not sure if you want to but might be handy to also add these placeholders in - https://github.com/DEFRA/cdp-node-frontend-template/commit/41c05beb62060159fbada94e1ca41d8e805b069a

**Pull request merged**